frontend/src/metabase/visualizations/click-actions/components/TimeseriesModeFooter.tsx

```diff
@@ -1,27 +1,46 @@
 import { t } from "ttag";
 import type { ModeFooterComponentProps } from "metabase/visualizations/types";
 import type Question from "metabase-lib/Question";
+import StructuredQuery from "metabase-lib/queries/StructuredQuery";
 import { TimeseriesFilterWidget } from "./TimeseriesFilterWidget";
 import { TimeseriesGroupingWidget } from "./TimeseriesGroupingWidget";
 
 type Props = ModeFooterComponentProps;
 
-export const TimeseriesModeFooter = (props: Props): JSX.Element => {
+export const TimeseriesModeFooter = (props: Props): JSX.Element | null => {
   const onChange = (question: Question) => {
     const { updateQuestion } = props;
     updateQuestion(question, { run: true });
   };
 
+  // We could encounter stale `mode` e.g. when converting a question from GUI to native,
+  // The `mode` would remain `timeseries` when it should have been `native` instead.
+  // So we shouldn't assume we'll always get timeseries question here.
+  if (!(props.query instanceof StructuredQuery)) {
+    return null;
+  }
+  const [breakout] = props.query.breakouts();
+  if (!breakout) {
+    return null;
+  }
+  const shouldHideTimeseriesGroupingWidget = breakout
+    .dimension()
+    .isExpression();
```

Comment:
This is the fix for the bug https://github.com/metabase/metabase/pull/33810

Reply:
Could you check for the list of subdimensions instead? E.g. `dimension.dimensions()` to have more than 1 item

Reply:
That doesn't work in this particular case aka. custom date column since I can still see a lot of subdimension from the custom columns

<img width="940" alt="Screenshot 2023-09-27 at 5 57 01 PM" src="https://github.com/metabase/metabase/assets/1937582/85161b5b-89f3-4535-87e3-86536e819049">

I initially tried that and it also failed the unit tests I added in `frontend/src/metabase/query_builder/containers/QueryBuilder.unit.spec.tsx`, so that solution doesn't work for this specific case.